app/components/candidate_interface/new_reference_history_component.html.erb

```diff
@@ -1,8 +1,8 @@
-<% history.each do |event| %>
+<% history.each.with_index do |event, i| %>
   <p class="govuk-body-s govuk-!-margin-top-2 govuk-!-margin-bottom-0">
-    <%= formatted_title event %> on <%= event.time.to_fs(:govuk_date) %>
+    <%= formatted_title(event) %> on <%= event.time.to_fs(:govuk_date) %>
 
-    <% if event.name == 'request_sent' %>
+    <% if event.name == 'request_sent' && can_be_cancelled(history[i+1]) %>
```

Comment:
I think would be more reliable if we could assert against the reference and not the history object.

Also, we could also move the whole condition to a method in the component? 

Something like (fake example):

```
   <% if can_be_cancelled? %>
```

And then on the component, you could use the `@reference` to verify if the reference can be canceled or not.

Happy to hear other people's thoughts.

Reply:
Agree, it's more legible and more reliable to use the state of the reference itself.

Reply:
Thanks guys, fixed in ac6d0de